delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/domain/Delivery.java

```diff
@@ -0,0 +1,72 @@
+package nl.softwarestrijders.waiter.delivery.core.domain;
+
+import nl.softwarestrijders.waiter.delivery.core.common.Generated;
+import nl.softwarestrijders.waiter.delivery.core.domain.exceptions.InvalidStatusUpdateException;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Class that contains the information about the Delivery
+ */
+@Document
+public class Delivery {
+    @Id
+    private UUID id;
+    private final DeliveryAddress address;
+    private Status status;
+    private final UUID orderId;
+
+    /**
+     * Constructor of {@link Delivery} class.
+     *
+     * @param address The address where the order needs te be delivered.
+     * @param status  The status of the delivery.
+     * @param orderId The Id of the order related to the delivery.
+     */
+    public Delivery(DeliveryAddress address, Status status, UUID orderId) {
+        Objects.requireNonNull(address, "Need an address to be able to deliver the order");
+        this.address = address;
+        setStatus(status);
+        Objects.requireNonNull(orderId, "Cannot make a delivery without an order reference");
+        this.orderId = orderId;
```

Comment:
You're doing duplicate things here, it would be better to just use the setters in here and leaving all the other logic in the setter

```suggestion
        Objects.requireNonNull(orderId, "Cannot make a delivery without an order reference");
        this.orderId = orderId;
        
        this.setStatus(status);
```